main.js

```diff
@@ -53,48 +58,90 @@ for (let i = 0; i < arr3.length; i++) {
   }
 }
 
-//lesson-3
-//1.
-function palindrome(str) {
-  str = str.toLowerCase();
-  return str === str.split('').reverse().join('');
-}
+// lesson-3
+// 1.
+const palindrome = (str) => {
+  const reverseStr = str.toLowerCase().split('').reverse().join('');
+  return str.toLowerCase() === reverseStr;
+};
 console.log(palindrome('Анна'));
+console.log(palindrome('mom'));
 
-//2.
-//min
+// 2.
+// min
 function min(a, b) {
-  return a > b ? a : b;
+  return a < b ? a : b;
 }
 console.log(min(2, 4));
 
-function min(a, b) {
-  return a < b ? a : b;
+function minTern(a, b) {
+  if (a < b) {
+    return a;
+  }
+  return b;
 }
-console.log(min(-1, -2));
+console.log(minTern(-1, -2));
 
-//max
+// max
 function max(a, b) {
   return a > b ? a : b;
 }
 console.log(max(200, 0));
 
-function max(a, b) {
-  return a > b ? a : b;
+function maxTern(a, b) {
+  if (a > b) {
+    return a;
+  }
+  return b;
 }
-console.log(max(-200, -201));
+console.log(maxTern(-200, -201));
 
-//3.
+// 3.
 function replaceZeros(array) {
+  const resultArr = [];
+
   for (let i = 0; i < array.length; i++) {
-    let numStr = '' + array[i];
+    let numStr = `${array[i]}`;
     if (numStr.indexOf('0') !== -1) {
       numStr = numStr.replaceAll('0', 'zero');
-      array[i] = numStr;
+      resultArr.push(numStr);
+    } else {
+      resultArr.push(`${array[i]}`);
     }
   }
+  return resultArr;
+}
+console.log(replaceZeros([100, 304, 5, 666, 909]));
+
+// lesson-4
+// 1.
+function someFunction(f) {
+  return function (a) {
+    return function (b) {
+      return f(a, b);
+    };
+  };
+}
+function getSum(a, b) {
+  return a + b;
```

Comment:
you have an unnecessary function wrapper "someFunction"

Reply:
TY. I'll fix it.